source/slides/13_ansible.rst

```diff
@@ -0,0 +1,520 @@
+  - name: restart apache
+    service: name=apache state=restarted
+
+Roles
+-----
+
+* Reusable list of tasks
+* Usually has a single goal (i.e. deploy apache)
+* Reusable
+
+.. code-block:: yaml
+
+  - hosts: demo
+    gather_facts: False
+    connection: local
+    serial: 1
+    vars:
+     in_ports:
+     - 80
+    roles:
+    - httpd
+    - mysql
+    - iptables
+
+Templates
+---------
+
+* Jinas2 templating engine
+* Use of variables
+* Loops, conditionals, filters, etc
+
+.. code-block:: jinja
+
+  # Build an apache Proxy config
+  < Proxy balancer://{{ balancer_name }}>
+  {% for host in groups['demo-web'] %}
+    BalancerMember http://{{ hostvars[host].ansible_eth1.ipv4.address }}
+  {% endfor %}
+    Order allow,deny
+    Allow from all
+  < /Proxy>
+
+``ansible-playbook``
+--------------------
+
+* Execute a playbook
+* Setting up a whole environment or set of hosts
+
+::
+
+  Usage: ansible-playbook playbook.yml -i inventory
+
+Best Practices
+--------------
+
+.. code-block:: bash
+
+  production      # inventory file for production servers
+  stage           # inventory file for stage environment
+
+  group_vars/
+     group1       # here we assign variables to particular groups
+     group2       # ""
+  host_vars/
+     hostname1    # if systems need specific variables, put them here
+     hostname2    # ""
+
+  site.yml        # master playbook
+  webservers.yml  # playbook for webserver tier
+  dbservers.yml   # playbook for dbserver tier
+
+  roles/
+      common/             # this hierarchy represents a "role"
+          tasks/          #
+              main.yml    #  <-- tasks file can include smaller files if warranted
+          handlers/       #
+              main.yml    #  <-- handlers file
+          templates/      #  <-- files for use with the template
+                          #      resource
+              ntp.conf.j2 #  <------- templates end in .j2
+          files/          #
+              bar.txt     #  <-- files for use with the copy resource
+              foo.sh      #  <-- script files for use with the script
+                          #      resource
+          vars/           #
+              main.yml    #  <-- variables associated with this role
+
+      webtier/            # same kind of structure as "common" was
+                          # above, done for the webtier role
+      monitoring/         # ""
+      fooapp/             # ""
+
+Variables
+---------
+
+* `Variables`__ allow you to deal with differences between systems
+* Valid names should be letters, numbers, and underscores and always start with
+  a letter
+
+.. rst-class:: codeblock-sm
+
+.. code-block:: yaml
+  :caption: playbook.yml
+
+  - hosts: webservers
+    template: src=foo.cfg.j2 dest={{ remote_install_path }}/foo.cfg
+    vars:
+      http_port: 80
+
+.. code-block:: jinja
+  :caption: Templates
+
+  My amp goes to {{ max_amp_value }}
+
+.. __: http://docs.ansible.com/ansible/playbooks_variables.html
+
+YAML Gotchas
```

Comment:
I think having this section is good.  The YAML gotchas can be annoying, unless you are already familiar with YAML or have an editor that groks YAML syntax and will provide assistance (still, this is not a cure all).